src/flb_sds.c

```
@@ -403,13 +403,14 @@ int flb_sds_snprintf(flb_sds_t *str, size_t size, const char *fmt, ...)
  retry_snprintf:
     va_start(va, fmt);
     ret = vsnprintf(*str, size, fmt, va);
-    if (ret > size) {
-        tmp = flb_sds_increase(*str, ret-size);
+    if (ret >= size) {
+        /* +1 for NUL terminator; vsnprintf writes at most size-1 chars. */
+        tmp = flb_sds_increase(*str, ret - size + 1);
         if (tmp == NULL) {
             return -1;
```

Cppcheck (2.20.0)

[error] va_list 'va' was opened but not closed by va_end().

(va_end_missing)